tests/kernel/run_ltp.pm

```diff
@@ -0,0 +1,303 @@
+    close $rfh;
+}
+
+sub parse_runfile {
+    my ($self, $cmd_file_text, $cmd_file) = @_;
+    my $cmd_pattern   = $_[-1] || '.*';
+    my @tests         = ();
+    my $tests_ignored = 0;
+
+    my ($result, $rfh) = $self->start_result('runfile', 'parse runfile');
+    say $rfh "## Parsing `$cmd_file` for tests which match `$cmd_pattern`";
+
+    for my $line (split(qr/[\n\r\f]+/, $cmd_file_text)) {
+        if ($line =~ /(^#)|(^$)/) {
+            next;
+        }
+
+        #Command format is "<name> <command> [<args>...] [#<comment>]"
+        if ($line =~ /^\s* (\w+) \s+ (\S.+) #?/gx) {
+            my $test = {name => $1, command => $2};
+            if ($test->{name} =~ $cmd_pattern) {
+                push @tests, $test;
+            }
+            else {
+                $tests_ignored++;
+            }
+            next;
+        }
+
+        if ($result->{result} ne 'fail') {
+            say $rfh "## Some lines could not be parsed: ";
+            $result->{result} = 'fail';
+            $self->{result}   = 'fail';
+        }
+        print $rfh $line;
+
+    }
+
+    say $rfh "\nUsing " . @tests . " test commands.";
+    print $rfh "Ignoring $tests_ignored test commands." if $tests_ignored > 0;
+    if (@tests < 1) {
+        $result->{result} = 'fail';
+        $self->{result}   = 'fail';
+    }
+
+    $self->commit_result($result, $rfh);
+
+    return @tests;
+}
+
+sub parse_openposix_runfile {
+    my ($self, $cmd_file_text, $cmd_pattern) = @_;
+
+    my ($result, $rfh) = $self->start_result('runfile', 'parse openposix runfile');
+    say $rfh 'Parsing Openposix runfile for tests which match ' . $cmd_pattern;
+
+    my @tests = ();
+    my @lines = split qr/[\n\r\f]+/, $cmd_file_text;
+    for my $line (@lines) {
+        if ($line =~ m/$cmd_pattern/) {
+            push @tests,
+              {
+                name    => basename($line, '.run-test'),
+                command => $line
+              };
+        }
+    }
+
+    print $rfh 'Using ' . @tests . ' of ' . @lines . ' Openposix test cases';
+    if (@tests < 1) {
+        $result->{result} = 'fail';
+        $self->{result}   = 'fail';
+    }
+
+    $self->commit_result($result, $rfh);
+
+    return @tests;
+}
+
+sub parse_ltp_log {
+    my ($self, $test_log, $fin_msg, $fh) = @_;
+    my $ignored_lines = 0;
+    my ($tconf, $tfail) = (0, 0);
+
+    for (split(qr/\n/, $test_log)) {
+        if (m/^\s+$/) {
+            next;
+        }
+        elsif (m/^(\w+)\s+(\d+)\s+(\w+)\s+:\s+(.*)$/) {
+            if ($3 eq 'TFAIL') {
+                $tfail = 1;
+                print $fh $_;
+            }
+            elsif ($3 eq 'TCONF') {
+                $tconf = 1;
+                print $fh $_;
+            }
+        }
+        elsif (m/$fin_msg(\d+)/) {
+            if ($1 != 0) {
+                say $fh "Test process returned none zero value ($1).";
+                unless ($tconf) {
+                    $tfail = 1;
+                }
```

Comment:
I would trust the exit value more than the test output here, since the LTP upstream script works that way as well.

So primarily set the `$tfail/$tconf` based on the exit value, 32 means `$tconf` and anything else non-zero could be mapped to `$tfail`. And then complain loudly and fail the test if the exit value is zero but the test output included any of `TFAIL`, `TBROK`, `TWARN` or `TCONF`, which would catch misbehaving tests.